docs/git/index.md

```diff
@@ -0,0 +1,220 @@
+
+4. Press the Clone button. Github Desktop should soon show you a screen like this:
+
+TODO: Screenshot
+
+5. Click the "Open in Visual Studio Code" button (TODO: Explain to Linux users how to do this, because they probably don't have the button).
+
+6. Visual Studio code should open, and you should see an "Explorer" bar which says `CYF-DEMO-REPO` and under it `README.md`. If you click on where it says `README.md`, it should open the file. Congratulations, you have just cloned your first Git repository!
+
+## What did we just do?
+
+TODO: Diagram showing a teacher's copy, a copy on Github, and a copy on each of three students' laptops.
+
+Each of you just cloned a repository which CodeYourFuture created onto your computer, and opened up your copy of one of the files. This is the process you're going to follow to get your homework every week.
+
+## Let's do some homework (teacher-led demo)
+
+The file you opened, `README.md`, has a question in it. We're going to answer the question, and submit it as if it was our homework.
+
+We can edit the file in VS Code to answer the question, and save it like normal.
+
+After we've saved the file, if we open up Github Desktop, something interesting has changed. On the left, it now says "1 changed file", and on the right, it shows us what the change was. Things we've removed get a red background, and things we've added get a green background.
+
+This is a really useful way for us to check over our homework before submitting it. If we've accidentally deleted things, or changed things we didn't mean to, we can notice now, and un-do them by editing the file again.
+
+When we're happy with our change, we can press the "Commit to main" button. That tells Git "This change is a good change, I want to keep it".
+
+Now, because we committed a change, our copy of the repository is different from CodeYourFuture's copy. Github Desktop now gives us a new button: "Push origin" ("origin" is what Git calls "where I cloned this repository from"). If we press it, it will try to send our change to the CodeYourFuture version.
+
+We didn't have this button before, because we hadn't committed any changes. There's an important distinction here: When we save files in VS Code, we store them in the file on our computer, but Git doesn't automatically commit them. It notices the changes (it showed them to us!), and asks us if we want to commit them. TODO: I'm not in love with any ways of describing "when to commit"...
+
+So, we're happy with our homework, we've committed it, let's try pushing it!
+
+## Forking
+
+Github Desktop just gave us a really useful warning, but it uses some words we haven't seen yet!
+
+We just tried to push our changes to CodeYourFuture's copy of this repository. But we're not allowed to do that! Imagine if anyone could just push anything they wanted to our repository! Then someone naughty could delete the homework, or worse.
```

Comment:
From @40thieves:

I like this explanation, but maybe we could frame it as "we don't want to push the answers into the CYF repo".

"Something naughty" is ambiguous and puts pushing in a negative light: "what if my push is accidentally naughty?", which I don't think is the intention.

Reply:
Good call - will reframe as accidental!